sqlalchemy-stubs/ext/asyncio/session.pyi

```diff
@@ -59,47 +77,93 @@ class AsyncSession:
         populate_existing: bool = ...,
         with_for_update: Optional[Any] = ...,
         identity_token: Optional[Any] = ...,
-    ): ...
+    ) -> Any: ...
     async def stream(
         self,
         statement: Any,
-        params: Optional[Any] = ...,
-        execution_options: Any = ...,
-        bind_arguments: Optional[Any] = ...,
+        params: Optional[
+            Union[Mapping[str, Any], Sequence[Mapping[str, Any]]]
+        ] = ...,
+        execution_options: _ExecutionOptions = ...,
+        bind_arguments: Optional[Mapping[str, Any]] = ...,
         **kw: Any,
-    ): ...
-    async def merge(self, instance: Any, load: bool = ...): ...
+    ) -> AsyncResult: ...
+    async def delete(self, instance: Any) -> None: ...
+    async def merge(self, instance: _T, load: bool = ...) -> _T: ...
     async def flush(self, objects: Optional[Any] = ...) -> None: ...
-    async def connection(self): ...
-    def begin(self, **kw: Any): ...
-    def begin_nested(self, **kw: Any): ...
-    async def rollback(self): ...
-    async def commit(self): ...
-    async def close(self): ...
+    async def connection(self) -> AsyncConnection: ...
+    def begin(self, **kw: Any) -> AsyncSessionTransaction: ...
+    def begin_nested(self, **kw: Any) -> AsyncSessionTransaction: ...
+    async def rollback(self) -> None: ...
+    async def commit(self) -> None: ...
+    async def close(self) -> None: ...
     @classmethod
-    async def close_all(self): ...
-    async def __aenter__(self): ...
+    async def close_all(self) -> None: ...
+    async def __aenter__(self: _TAsyncSession) -> _TAsyncSession: ...
     async def __aexit__(
         self, type_: Any, value: Any, traceback: Any
     ) -> None: ...
+    # copied via create_proxy_methods
```

Comment:
can't we create like a mock superclass (or protocol) just in the typing that has these methods and is shared by both the session and the async one (and the scoped session also)

Reply:
I think that's a good idea, so I'll go ahead and implement that

Reply:
Thinking about this a bit more, I think we (me, most likely) should implement this change after all of the current PRs are landed so as to reduce changes between files right now.

Reply:
sure, it makes sense. I've added https://github.com/sqlalchemy/sqlalchemy2-stubs/issues/61 as a reminder